File: Arrays/Remove_Duplicate/remove-duplicates.c
#include "main.h"

int removeDuplicates(int* nums, int numsSize){
	int i, j, count = 0;
	if (numsSize < 2)
		return (numsSize);
	for (i = 0; i < numsSize - 1; i++)
	{
		if (nums[i + 1] == nums[i])
		{
			for (j = i; j < numsSize - 1; j++)
				nums[j] = nums[j + 1];
			nums[numsSize - 1] = 0;
			numsSize--;
			i--, count--;
		}
		count++;
	}
	return (++count);
}

File: Arrays/Remove_Duplicate/remove-duplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var removeDuplicates = function(nums) {
	let i, j, count = 0;
	for (i = 0; i < nums.length; i++)
	{
		if (nums[i + 1] == nums[i])
		{
			for (j = i; j < nums.length - 1; j++)
				nums[j] = nums[j + 1];
			nums.pop();
			i--, count--;
		}
		count++;
	}
	return (count);
};
